### PR TITLE
Update FAQs

### PR DESCRIPTION
**Pull request by Joyinqin**
### What is changed, added or deleted? (Required)
Update out of date FAQs.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?
[SQL FAQs](https://docs.pingcap.com/tidb/stable/sql-faq); [Deploy and Maintain FAQs](https://docs.pingcap.com/tidb/stable/deploy-and-maintain-faq)

- This PR is translated from: https://github.com/pingcap/docs/pull/3153
- Other reference link(s):

### CODE REVIEW

**Comment by Joyinqin** [comment]
@yikeke PTAL

File: faq/deploy-and-maintain-faq.md [hunk 1]
```diff
@@ -340,7 +340,7 @@ TiClient Region Error 该指标描述的是在 TiDB-server 作为客户端通过
 
 #### TiDB 同时支持的最大并发连接数？
 
-当前版本 TiDB 没有最大连接数的限制，如果并发过大导致响应时间增加，可以通过增加 TiDB 节点进行扩容。
+默认情况下，每个 TiDB 服务器的最大连接数没有限制。可以通过在 `config.toml` 文件中设置 `max-server-connections` 来实施限制。如果并发量过大导致响应时间增加，建议通过添加 TiDB 节点进行扩容。
```
Comment (lilin90) [inline faq/deploy-and-maintain-faq.md:343]:
```suggestion
默认情况下，每个 TiDB 服务器的最大连接数没有限制。如有需要，可以在 `config.toml` 文件中设置 `max-server-connections` 来限制最大连接数。如果并发量过大导致响应时间增加，建议通过添加 TiDB 节点进行扩容。
```

File: faq/sql-faq.md [hunk 2]
```diff
@@ -191,7 +191,7 @@ RUNNING_JOBS: ID:121, Type:add index, State:running, SchemaState:write reorganiz
 
 ### TiDB 是否支持基于 COST 的优化（CBO），如果支持，实现到什么程度？
 
-是的，TiDB 使用的基于成本的优化器（CBO），我们有一个小组单独会对代价模型、统计信息持续优化，除此之外，我们支持 hash join、soft merge 等关联算法。
+是的，TiDB 使用的基于成本的优化器（CBO），我们有一个小组单独会对代价模型、统计信息持续优化，除此之外，我们支持 hash join、soft-merge 等 join 算法。
```
Comment (lilin90) [inline faq/sql-faq.md:194]:
```suggestion
是的，TiDB 使用的是基于成本的优化器 (CBO)，会对代价模型、统计信息持续优化。除此之外，TiDB 还支持 hash join、soft-merge join 等 join 算法。

```

File: faq/sql-faq.md [hunk 3]
```diff
@@ -76,7 +76,7 @@ TiDB 的 `show processlist` 与 MySQL 的 `show processlist` 显示内容基本
 
 + 由于 TiDB 是分布式数据库，TiDB server 实例是无状态的 SQL 解析和执行引擎（详情可参考 [TiDB 整体架构](/tidb-architecture.md)），用户使用 MySQL 客户端登录的是哪个 TiDB server ，`show processlist` 就会显示当前连接的这个 TiDB server  中执行的 session 列表，不是整个集群中运行的全部 session 列表；而 MySQL 是单机数据库，`show processlist` 列出的是当前整个 MySQL 数据库的全部执行 SQL 列表。
 
-+ TiDB 的 `show processlist` 显示内容比起 MySQL 来讲，多了一个当前 session 使用内存的估算值（单位 Byte）。
++ 在查询执行期间，TiDB 中的 `State` 列不会持续更新。由于 TiDB 支持并行查询，每个语句可能同时处于多个状态，因此很难简化为单个值。
```
Comment (zz-jason) [inline faq/sql-faq.md:79]: > 由于 TiDB 支持并行查询，每个语句可能同时处于多个状态，因此很难简化为单个值。

什么很难简化为单个值？

Comment (Joyinqin) [reply faq/sql-faq.md:79]: 状态很难简化为单个值，这样表述对吗？

Comment (lilin90) [inline faq/sql-faq.md:79]:
```suggestion
+ 在查询执行期间，TiDB 中的 `State` 列不会持续更新。由于 TiDB 支持并行查询，每个语句可能同时处于多个状态，因此很难显示为某一种状态。
```

Comment (Joyinqin) [reply faq/sql-faq.md]: 已更改。@zz-jason PTAL

**Comment by lilin90** [comment]
@Joyinqin Please fix the lint failure.

**Comment by lilin90** [comment]
@Joyinqin Since the organization of FAQs in v3.1, v3.0, v2.1 is quite different from that of dev and v4.0, I've closed the cherry-picked PRs to those three branches. **Please create another PR to the `release-3.1` branch according to https://github.com/pingcap/docs/pull/3153 and then cherry pick to v3.0 and v2.1 in that PR.**

**Pull request merged**